src/Estimators/include/BipedalLocomotion/Estimators/FloatingBaseEstimator.h

```diff
@@ -0,0 +1,274 @@
+    *  All the model related kinematics and dynamics computations specific to
+    *  the floating base estimator are contained within this class.
+    */
+    class ModelComputations
+    {
+    public:
+        /**
+        * Set the reduced model
+        * @param[in] model reduced iDynTree model
+        * @return True in case of success, false otherwise.
+        */
+        bool setModel(const iDynTree::Model& model);
+
+        /**
+        * Set base link name and name of the IMU rigidly attached to the IMU
+        * @param[in] base_link_frame name of the base link frame
+        * @param[in] imu_frame name of the IMU frame
+        * @return True in case of success, false otherwise.
+        */
+        bool setBaseLinkAndIMU(const std::string& base_link_frame, const std::string& imu_frame);
+
+        /**
+        * Set the feet contact frames, expected to be in contact with the environment
+        * @param[in] l_foot_contact_frame left foot contact frame
+        * @param[in] r_foot_contact_frame right foot contact frame
+        * @return True in case of success, false otherwise.
+        */
+        bool setFeetContactFrames(const std::string& l_foot_contact_frame, const std::string& r_foot_contact_frame);
+
+        /**
+        * Check if model is configured with the required information
+        * @note this is a required step to initialize the estimator
+        * @return True in case of success, false otherwise.
+        */
+        bool checkModelInfoLoaded();
+
+        /**
+        * Get relative pose between IMU and the feet
+        * @param[in] encoders joint positions through encoder measurements
+        * @param[out] IMU_H_l_foot pose of the left foot contact frame with respect to the IMU frame
+        * @param[out] IMU_H_r_foot pose of the right foot contact frame with respect to the IMU frame
+        * @return True in case of success, false otherwise.
+        */
+        bool getIMU_H_feet(const iDynTree::JointPosDoubleArray& encoders,
+                           iDynTree::Transform& IMU_H_l_foot,
+                           iDynTree::Transform& IMU_H_r_foot);
+
+        /**
+        * Get the base link pose and velocity from the estimated IMU pose and velocity
+        * @note the input and output velocities are both specified in mixed-velocity representation
+        * @param[in] A_H_IMU pose of the IMU in the world
+        * @param[in] v_IMU mixed-trivialized velocity of the IMU in the world
+        * @param[out] A_H_B pose of the base link in the world
+        * @param[out] v_B mixed-trivialized velocity of the base link in the world
+        * @return True in case of success, false otherwise.
+        */
+        bool getBaseStateFromIMUState(const iDynTree::Transform& A_H_IMU, const iDynTree::Twist& v_IMU,
+                                      iDynTree::Transform& A_H_B, iDynTree::Twist& v_B);
+
+        /**
+         * Getters
+         */
+        const auto& nrJoints() const { return m_nr_joints; }
```

Comment:
I would suggest not using ``auto`` as output type of a public method in a header.

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/79/commits/29148cbdbdb7d9e43e8b50c5434b0daeb7af4706